src/ios/InputTransformer/OSFANLInputTransformer.swift

```diff
@@ -0,0 +1,88 @@
+struct OSFANLInputTransformer: OSFANLInputTransformable {
+    func transform(_ eventParameterArray: [InputParameterData]?, _ itemArray: [InputItemData]?) throws -> OSFANLInputTransformableModel {
+        var eventParameterData: InputParameterData?
+        if let eventParameterArray {
+            guard let flatResult = try? self.flat(keyValueMapArray: eventParameterArray) else {
+                throw OSFANLError.duplicateItemsIn(parameter: OSFANLInputDataFieldKey.eventParameters.rawValue)
+            }
+            eventParameterData = flatResult
+        }
+        let itemArray = try self.transform(itemArray)
+        
+        return .init(eventParameterData, itemArray)
+    }
+}
+
+private extension OSFANLInputTransformer {
+    func flat(keyValueMapArray array: [InputParameterData]) throws -> InputParameterData {
+        let flatKeyValueArray = array.map {
+            [$0[OSFANLInputDataFieldKey.key.rawValue, default: ""]: $0[OSFANLInputDataFieldKey.value.rawValue, default: ""]]
+        }
+        let flatKeyValueDictionary = self.flat(dictionaryArray: flatKeyValueArray)
+        
+        if flatKeyValueDictionary.count != flatKeyValueArray.count { throw OSFANLError.duplicateKeys }
+        return flatKeyValueDictionary
+    }
+    
+    func transform(_ itemArray: [InputItemData]?) throws -> [InputItemData]? {
+        // if nil or empty, just return it as there's nothing to do here.
+        guard let itemArray, !itemArray.isEmpty else {
+            return itemArray
+        }
+        // if there's more than the expected maximum, return the error
+        guard itemArray.count <= OSFANLDefaultValues.eventItemsMaximum else {
+            throw OSFANLError.tooMany(parameter: OSFANLInputDataFieldKey.items.rawValue, limit: OSFANLDefaultValues.eventItemsMaximum)
+        }
+        
+        var resultArray: [InputItemData] = []
+        for itemData in itemArray {
+            // there needs to be at least one `item_id` or `item_name`
+            guard itemData.keys.contains(where: { [OSFANLInputDataFieldKey.itemId, .itemName].map(\.rawValue).contains($0) }) else {
+                throw OSFANLError.missingItemIdName
+            }
+            
+            var resultItem: InputItemData = [:]
+            // insert quantity when missing
+            if !itemData.keys.contains(OSFANLInputDataFieldKey.quantity.rawValue) {
+                resultItem[OSFANLInputDataFieldKey.quantity.rawValue] = OSFANLDefaultValues.itemQuantity
```

Comment:
Isn't this already taken care of by the SDK? According to https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtag#add_payment_info_item, if not set, `quantity` is already set to `1`.

Reply:
You're right. And this could also be applied to some other validations as well. The thing is that I wasn't able to fetch the item array from the Analytics Dashboard so decided to add this as a fail-safe situation. 

I'll have a second look into this but I agree with you. If we're trusting the SDK's documentation, this is not necessary.

Reply:
Ohh I see, the Analytics Dashboard can be tricky 😅 .

What you said makes sense for me though, we could keep it as a fail-safe measure, and the code is actually pretty simple. My comment was more of a suggestion really, and so this shouldn't block the merge.